### ISSUE TITLE
Time measure units returning null

### ISSUE THREAD

**Issue by g3kr** [spencermountain/compromise#558]
Any time measure in the singular format returns 'null' unit

last 1 day
last 1 year
last 1 minute
last 1 hour

Code:
```
const doc = nlp(‘last 1 year')
console.log(doc.values().data())
```

Output
```
[ { number: 1,
    nice: '1',
    ordinal: '1st',
    niceOrdinal: '1st',
    text: 'one',
    textOrdinal: 'first',
    unit: '' } ]
```

I would expect it to return years or year. Is this expected?

**Comment by spencermountain** [comment]
hey @g3kr, yeah you're right.
The unit thing is pretty half-baked to be honest, and needs some major re-thinking.
it also clashes like this, with dates.
future versions of compromise I think, should address this in a much different way.

**Comment by g3kr** [comment]
Thanks for the quick response. Is this something that would be easy to add for the time-being or you are thinking about a whole different approach? 

I can see that other units like weight, behaves as expected. `nlp('1 gram').values().data()` can classify the gram as unit.

**Comment by spencermountain** [comment]
yeah, been [working on a full re-write](https://github.com/spencermountain/compromise/tree/linked-list) and a [separate date-parsing library](https://github.com/spencermountain/half-past)

neither is ready, and will take a few months.

**Comment by spencermountain** [comment]
oh i'm wrong, i think this is an issue about the singular noun.

```
let doc = nlp('7 years')
console.log(doc.values().data().unit)
// 'years'
let doc = nlp('1 year')
console.log(doc.values().data().unit)
// 'year'
```

**Comment by g3kr** [comment]
Thanks for taking a look. 

This works for `1 year` or `1 day`. However if you mistyped `5 day` instead of `5 days` it seems to break. Is this expected?
